### PR TITLE
Deps upgrade erl 22

### PR DESCRIPTION
**Pull request by telezynski**
This PR is to documment the current state of deps upgrades.
The following errors in tests were observed:

```TRAVIS OTP 22

~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6162: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 1 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6258: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 2 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6354: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 3 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6518: Testing big_tests.tests.connect_SUITE: *** FAILED test case 42 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6588: Testing big_tests.tests.connect_SUITE: *** FAILED test case 43 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6676: Testing big_tests.tests.connect_SUITE: *** FAILED test case 49 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6746: Testing big_tests.tests.connect_SUITE: *** FAILED test case 50 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6834: Testing big_tests.tests.connect_SUITE: *** FAILED test case 56 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:6904: Testing big_tests.tests.connect_SUITE: *** FAILED test case 57 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:7694: Testing big_tests.tests.connect_SUITE: *** FAILED test case 73 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:7696: Testing big_tests.tests.connect_SUITE: *** FAILED test case 74 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:8280: Testing big_tests.tests.connect_SUITE: *** FAILED test case 82 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:8282: Testing big_tests.tests.connect_SUITE: *** FAILED test case 83 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:9086: Testing big_tests.tests.connect_SUITE: *** FAILED test case 91 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:9088: Testing big_tests.tests.connect_SUITE: *** FAILED test case 92 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:12266: Testing big_tests.tests.sm_SUITE: *** FAILED test case 22 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:12700: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 15 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:12826: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 16 ***
~/Downloads/TRAVIS_log_2_mnesia_TLS.txt:12952: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 17 ***

----

~/Downloads/TRAVIS_log_2_mssql_odbc.txt:6991: Testing big_tests.tests.connect_SUITE: *** FAILED test case 42 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:7061: Testing big_tests.tests.connect_SUITE: *** FAILED test case 43 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:7149: Testing big_tests.tests.connect_SUITE: *** FAILED test case 49 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:7219: Testing big_tests.tests.connect_SUITE: *** FAILED test case 50 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:7307: Testing big_tests.tests.connect_SUITE: *** FAILED test case 56 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:7377: Testing big_tests.tests.connect_SUITE: *** FAILED test case 57 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:8167: Testing big_tests.tests.connect_SUITE: *** FAILED test case 73 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:8169: Testing big_tests.tests.connect_SUITE: *** FAILED test case 74 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:8911: Testing big_tests.tests.connect_SUITE: *** FAILED test case 82 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:8935: Testing big_tests.tests.connect_SUITE: *** FAILED test case 83 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:9609: Testing big_tests.tests.connect_SUITE: *** FAILED test case 91 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:9643: Testing big_tests.tests.connect_SUITE: *** FAILED test case 92 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:13589: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 15 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:13713: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 16 ***
~/Downloads/TRAVIS_log_2_mssql_odbc.txt:13839: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 17 ***

----

~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6328: Testing big_tests.tests.connect_SUITE: *** FAILED test case 42 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6398: Testing big_tests.tests.connect_SUITE: *** FAILED test case 43 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6486: Testing big_tests.tests.connect_SUITE: *** FAILED test case 49 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6556: Testing big_tests.tests.connect_SUITE: *** FAILED test case 50 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6644: Testing big_tests.tests.connect_SUITE: *** FAILED test case 56 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:6714: Testing big_tests.tests.connect_SUITE: *** FAILED test case 57 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:7638: Testing big_tests.tests.connect_SUITE: *** FAILED test case 73 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:7640: Testing big_tests.tests.connect_SUITE: *** FAILED test case 74 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:8218: Testing big_tests.tests.connect_SUITE: *** FAILED test case 82 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:8220: Testing big_tests.tests.connect_SUITE: *** FAILED test case 83 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:8780: Testing big_tests.tests.connect_SUITE: *** FAILED test case 91 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:8782: Testing big_tests.tests.connect_SUITE: *** FAILED test case 92 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:10322: Testing big_tests.tests.pubsub_SUITE: *** FAILED test case 8 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:11676: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 6 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:11808: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 29 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:11934: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 30 ***
~/Downloads/TRAVIS_log2_ldap_mnesia.txt:12060: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 31 ***


====

TRAVIS_21.3

~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:6111: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 1 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:6207: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 2 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:6303: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 3 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:10013: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 9 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:10145: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 29 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:10269: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 30 ***
~/Downloads/TRAVIS_21.3_mnesia_TLS.txt:10395: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 31 ***

----

~/Downloads/TRAVIS_21.3_mssql_odbc.txt:10905: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 15 ***
~/Downloads/TRAVIS_21.3_mssql_odbc.txt:11031: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 16 ***
~/Downloads/TRAVIS_21.3_mssql_odbc.txt:11157: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 17 ***

----

~/Downloads/TRAVIS_21.3_ldap_mnesia.txt:9188: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 15 ***
~/Downloads/TRAVIS_21.3_ldap_mnesia.txt:9312: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 16 ***
~/Downloads/TRAVIS_21.3_ldap_mnesia.txt:9438: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 17 ***

----

~/Downloads/TRAVIS_21.3_mysql_redis_amqp.txt:10912: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 15 ***
~/Downloads/TRAVIS_21.3_mysql_redis_amqp.txt:11038: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 16 ***
~/Downloads/TRAVIS_21.3_mysql_redis_amqp.txt:11164: Testing big_tests.tests.mod_global_distrib_SUITE: *** FAILED test case 17 ***

====```

### CODE REVIEW

**Comment by telezynski** [comment]
Errors in big_tests.tests.connect_SUITE under OTP 22 are (most probably) the result of:
`For security reasons TLS-1.1 is no longer supported by default, but can be configured. (OTP 22)`

**Comment by telezynski** [comment]
JIRA TICKET: https://erlangsolutions.atlassian.net/browse/MIM-385

List of dependencies that are outdated:
https://erlangsolutions.atlassian.net/wiki/spaces/MON/pages/393773057/List+of+dependencies+that+are+outdated

File: .travis.yml [hunk 1]
```diff
@@ -50,20 +50,21 @@ after_success:
         - test 1 = "$SKIP_COV" || ./rebar3 codecov analyze
         - test 1 = "$SKIP_COV" || codecov --disable=gcov --env PRESET
         - test 1 = "$SKIP_COV" || ./rebar3 coveralls send
-        - if [ -n "${DOCKERHUB_PASS}" ] && [ $PRESET = 'internal_mnesia' ] && [ $TRAVIS_OTP_RELEASE = "20.3" ];
```
Comment (fenek) [inline .travis.yml]: Changes in this file are only temporary, correct?

File: big_tests/rebar.config [hunk 2]
```diff
@@ -31,22 +31,6 @@
...
-        {fast_tls, ".*", {git, "https://github.com/processone/fast_tls", {tag, "1.0.17"}}},
-        {p1_utils, ".*", {git, "https://github.com/processone/p1_utils", {tag, "1.0.10"}}}
-    ]}]},
-   {override, fast_tls, [{plugins, [pc]}, {deps, [
-        {p1_utils, ".*", {git, "https://github.com/processone/p1_utils", {tag, "1.0.10"}}}
-    ]}]}
-
-  ]}.
+{overrides, [
+  {override, rebar3_codecov, [{deps, [{jiffy, "1.0.1"}]}]}
```
Comment (fenek) [inline big_tests/rebar.config]: What does it mean?

File: big_tests/tests/rest_helper.erl [hunk 3]
```diff
@@ -164,7 +164,7 @@ decode(RespBody) ->
     try
         jiffy:decode(RespBody)
     catch
-        throw:_ ->
+        error:_ ->
```
Comment (fenek) [inline big_tests/tests/rest_helper.erl]: Really? Is this the reason why REST tests were failing after `jiffy` upgrade? :D

File: src/mongoose_metrics.erl [hunk 4]
```diff
@@ -292,6 +292,33 @@ merge_stats_fun(_, V1, V2) ->
 inet_stats(Port) when is_port(Port) ->
     {ok, Stats} = inet:getstat(Port, ?INET_STATS),
     Stats;
+%% TLS dist is operated by PID, not PORT directly, so we need to find the relevant port
+inet_stats(Pid) when is_pid(Pid) ->
```
Comment (fenek) [inline src/mongoose_metrics.erl]: How is it related to deps upgrade?

File: .travis.yml (hunk 1 above)
Comment (telezynski) [reply .travis.yml]: yes, reverted just now

File: big_tests/rebar.config (hunk 2 above)
Comment (telezynski) [reply big_tests/rebar.config]: Jiffy is used in this rebar plugin (rebar3_codecov) as well and is hardcoded there as {vsn,"0.15.2"}.
I could have left it as it is, as it is not used directly in release nor in tests, however just to not have any confusion regarding version I decided to upgrade it as well.
The only way to upgrade a dependency of dependency is to use override option in rebar.config.
I think it is an explicit way of telling in main project what we want. The other way would be to ask maintainer of codecov plugin to change it.

File: big_tests/tests/rest_helper.erl (hunk 3 above)
Comment (telezynski) [reply big_tests/tests/rest_helper.erl]: yes, they changed the way they raise errors in decode

File: src/mongoose_metrics.erl (hunk 4 above)
Comment (telezynski) [reply src/mongoose_metrics.erl]: This is not related but came out as I was the only one to test MIM in OTP 22 and at first there was confusion which errors come from deps upgrade and which are related to OTP 22.
This issue happens on master of MIM with OTP 22. We could have a separate ticket for this but as we already had a ticket it went there.
The ticket with description of the issue (solved by Rafal Slota) is https://erlangsolutions.atlassian.net/browse/MIM-441

File: big_tests/rebar.config (hunk 2 above)
Comment (DenysGonchar) [reply big_tests/rebar.config]: it's ESL homemade plugging, update the version in the `rebar3_codecov` repo.
also let's not confuse anyone with extra override in our `rebar.config`

File: src/mongoose_metrics.erl (hunk 4 above)
Comment (DenysGonchar) [reply src/mongoose_metrics.erl]: Jira ticket mentions both OTP 21 and 22 [Fix OTP21,22/cluster_commands_SUITE](https://erlangsolutions.atlassian.net/browse/MIM-441)

is it OTP 22 specific? TLS dist. was working fine for OTP 21.3, i tested it less than a month ago

Comment (telezynski) [reply src/mongoose_metrics.erl]: It was failing for me on master with OTP 21.3 on 26th of June (no deps upgrade changes).

```
~/Downloads/TRAVIS_master_21.3_mnesia_TLS.txt:6114: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 1 ***
~/Downloads/TRAVIS_master_21.3_mnesia_TLS.txt:6210: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 2 ***
~/Downloads/TRAVIS_master_21.3_mnesia_TLS.txt:6306: Testing big_tests.tests.cluster_commands_SUITE: *** FAILED test case 3 ***

```

File: big_tests/rebar.config (hunk 2 above)
Comment (telezynski) [reply big_tests/rebar.config]: Interestingly it is not in ESL repo. As are several of ESL made deps in MIM.

Comment (DenysGonchar) [reply big_tests/rebar.config]: we can clone it to ESL. ask Karol for help, he has required rights to create new ESL repo.

**Pull request merged**